tests/onnx/test_features.py

```diff
@@ -0,0 +1,115 @@
+from tempfile import TemporaryDirectory
+from unittest import TestCase
+from unittest.mock import MagicMock, patch
+
+from transformers import AutoModel, TFAutoModel
+from transformers.onnx import FeaturesManager
+from transformers.testing_utils import SMALL_MODEL_IDENTIFIER, require_tf, require_torch
+
+
+@require_torch
+@require_tf
+class DetermineFrameworkTest(TestCase):
+    """
+    Test `FeaturesManager.determine_framework`
+    """
+
+    def setUp(self):
+        self.test_model = SMALL_MODEL_IDENTIFIER
+
+    def _setup_pt_ckpt(self, save_dir):
+        model_pt = AutoModel.from_pretrained(self.test_model)
+        model_pt.save_pretrained(save_dir)
+
+    def _setup_tf_ckpt(self, save_dir):
+        model_tf = TFAutoModel.from_pretrained(self.test_model, from_pt=True)
+        model_tf.save_pretrained(save_dir)
+
+    def test_framework_provided(self):
+        """
+        Ensure the that the provided framework is returned.
+        """
+        mock_framework = "mock_framework"
+
+        # Framework provided - return whatever the user provides
+        result = FeaturesManager.determine_framework(self.test_model, mock_framework)
+        self.assertEqual(result, mock_framework)
+
+        # Local checkpoint and framework provided - return provided framework
+        # PyTorch checkpoint
+        with TemporaryDirectory() as local_pt_ckpt:
+            self._setup_pt_ckpt(local_pt_ckpt)
+            result = FeaturesManager.determine_framework(local_pt_ckpt, mock_framework)
+            self.assertEqual(result, mock_framework)
+
+        # TensorFlow checkpoint
+        with TemporaryDirectory() as local_tf_ckpt:
+            self._setup_tf_ckpt(local_tf_ckpt)
+            result = FeaturesManager.determine_framework(local_tf_ckpt, mock_framework)
+            self.assertEqual(result, mock_framework)
+
+    def test_checkpoint_provided(self):
+        """
+        Ensure that the determined framework is the one used for the local checkpoint.
+
+        For the functionality to execute, local checkpoints are provided but framework is not.
+        """
+        torch_str = "pt"
```

Comment:
Nit: to save a bit of code duplication + ease of maintenance, we could move these variables to `setUp` and then access them with `self.torch_str`

We also tend to prefer putting the type on variable names, so an alternative would be to use something like `framework_pt = "pt"` and `framework_tf = "tf"` instead of `torch_str` etc

Reply:
Good idea, thanks! Done in [6bd7477](https://github.com/huggingface/transformers/pull/18615/commits/6bd7477fb2ddb5932ac46b0f59a945bfe3290d12).

Reply:
Thanks!